include/pmacc/particles/memory/dataTypes/SuperCell.hpp

```diff
@@ -77,6 +78,19 @@ namespace pmacc
 #endif
         {
             constexpr uint32_t frameSize = T_FrameType::frameSize;
+
+            /* NOTE on result expression understanding:
+             * (numParticles % frameSize) =^= how many particle did not fit in a not full frame?
+             *
+             * but we need how many are in the last frame
+             * => (numParticles - 1u) % frameSize + 1u
+             *   only shift by one which is reversed by + 1u
+             *  => will return the same result for numParticles =/= i * frameSize ;i \in N
+             *
+             *  and for ((frameSize * i) - 1u) % frameSize + 1u = (frameSize - 1u) + 1u
+             *  = frameSize; QED
```

Comment:
what is `QED`

Reply:
QED is an abbreviation for "quod erad demonstrandum", latin: "that which was to be demonstrated"